### PR TITLE
add c_api_chgmipstarts

### PR DESCRIPTION
**Pull request by remi-garcia**
add change_mip_start! and rename set_warm_start! to set_mip_start!

I did not know that optimize! removes the mip start so I have no use for this function but it is done anyway

### CODE REVIEW

File: src/cpx_model.jl [hunk 1]
```diff
@@ -224,6 +224,27 @@ function set_warm_start!(model::Model, indx::IVec, val::FVec, effortlevel::Integ
     end
 end
 
+change_mip_start!(model::CPLEX.Model, x::Vector{Float64}, effortlevel::Integer = CPX_MIPSTART_AUTO) = change_mip_start!(model, Cint[1:length(x);], x, effortlevel)
```
Comment (odow) [inline src/cpx_model.jl]: We're trying to standardize our naming of the low-level functions to `c_api_xxx`. See, e.g., https://github.com/JuliaOpt/CPLEX.jl/blob/cb0c9226f7a49aff5a2358b0c82a3ebb5151a340/src/cpx_constrs.jl#L1.

So this should be named `c_api_chgmipstarts`.

File: src/cpx_model.jl [hunk 2]
```diff
@@ -204,9 +204,9 @@ function c_api_chgobj(model::Model, indices::IVec, values::FVec)
     end
 end
 
-set_warm_start!(model::Model, x::Vector{Float64}, effortlevel::Integer = CPX_MIPSTART_AUTO) = set_warm_start!(model, Cint[1:length(x);], x, effortlevel)
+set_mip_start!(model::Model, x::Vector{Float64}, effortlevel::Integer = CPX_MIPSTART_AUTO) = set_mip_start!(model, Cint[1:length(x);], x, effortlevel)
```
[the remark this answers is not in the document]
Comment (remi-garcia) [reply src/cpx_model.jl]: Sorry, that should be obvious. ```@deprecate``` done.

File: src/cpx_model.jl (hunk 1 above)
Comment (remi-garcia) [reply src/cpx_model.jl]: I did not know for this standardization, I read the pull request from 2017, corrected now.

Comment (odow) [reply src/cpx_model.jl]: No worries. This wrapper is one of the older JuliaOpt packages. It could do with a big clean up.

**Comment by remi-garcia** [comment]
New test: test/C_API/mip_02.jl
I did not check the test since I have not CPLEX on this machine

**Comment by remi-garcia** [comment]
Can I add unrelated changes to this PR or should I do a specific PR for each type of change? I might need [CPXXgetdj](https://www.ibm.com/support/knowledgecenter/SSSA5P_12.8.0/ilog.odms.cplex.help/refcallablelibrary/cpxapi/getdj.html) so I'll add it.

**Comment by odow** [comment]
Please open new prs

**Comment by remi-garcia** [comment]
In fact CPXXgetdj was already present. Good to know anyway for the prs. Thanks

**Comment by remi-garcia** [comment]
I'll close this PR and start over. I did too many mistakes on that fork and need to do that better now that I've a better understanding of Github

**Pull request closed without merging**